packages/components/src/tooltip/index.tsx

```diff
@@ -96,9 +98,36 @@ function UnconnectedTooltip(
 	} );
 
 	if ( isNestedInTooltip ) {
+		// Avoid passing certain props added by the useContextSystem hook when
+		// cloning children, as they are not intended for this scenario.
+		const clonedProps = removeExtraPropsAddedByContext(
+			restProps,
+			'Tooltip'
+		);
+		const chainEvent =
+			( eventName: keyof typeof clonedProps ) =>
+			( ...args: unknown[] ): void => {
+				children.props[ eventName ]?.( ...args );
+				clonedProps[ eventName ]?.( ...args );
+			};
 		return isOnlyChild
 			? cloneElement( children, {
-					...restProps,
+					...clonedProps,
+					// Merge incoming classname with existing classname.
+					className: classNames(
+						children.props.className,
+						clonedProps.className
+					),
+					// Merge incoming inline styles with existing styles.
+					style: {
+						...children.props.style,
+						...clonedProps.style,
+					},
+					// Chain event listeners by calling both the existing and the
+					// incoming callback.
+					onClick: chainEvent( 'onClick' ),
+					onMouseEnter: chainEvent( 'onMouseEnter' ),
+					onBlur: chainEvent( 'onBlur' ),
```

Comment:
@diegohaz I'd like to get a check from your side that this is a good approach — basically making sure that, when we clone `children` in inner nested tooltips (ie. the tooltip part gets ignored), we forward the correct props to the anchor.

I started by merging `className` and `styles`, then I realized that I needed to remove a few extra things added automatically by the context system (see the `removeExtraPropsAddedByContext` function). Finally, I had to chain events, and for now I chose to chain the ones that seem essential to keep the `ariakit` tooltip functioning correctly.

Please let me know if this approach makes sense to you, or if you have suggestions on how to improve it 🙏

Reply:
> @diegohaz I'd like to get a check from your side that this is a good approach — basically making sure that, when we clone `children` in inner nested tooltips (ie. the tooltip part gets ignored), we forward the correct props to the anchor.

I'm not certain I grasp the intent behind that. Could you share a code example where this is used and it caused a problem when all that merge logic wasn't applied?

> I chose to chain the ones that seem essential to keep the `ariakit` tooltip functioning correctly.

This is particularly concerning. Those events aren't part of Ariakit's public API. They could be taken out and new _essential_ event handlers might be introduced in patch versions.

Reply:
I recorded a video to give more context:


https://github.com/WordPress/gutenberg/assets/1083581/09bd2a3c-7cd9-43a9-bd66-d26b6a07416d

Reply:
Thanks for the video; it was really helpful.

Could you try swapping `cloneElement` with `Ariakit.Role`? I did a quick run of this code in Storybook and it seems to be functioning as expected, though I might be overlooking something:

```jsx
		return isOnlyChild ? (
			<Ariakit.Role { ...clonedProps } render={ children } />
		) : (
			children
		);
```

This is essentially the same, but it automatically merges any function prop that starts with `on` (if conflicts arise). This should be a safer approach.

Reply:
Sounds good, thank you! I gave it a quick try and it seems to work as expected on my end too.